src/check_expr.cpp

```diff
@@ -1163,20 +1167,23 @@ gb_internal void check_assignment(CheckerContext *c, Operand *operand, Type *typ
 		switch (operand->mode) {
 		case Addressing_Builtin:
 			error(operand->expr,
-			      "Cannot assign built-in procedure '%s' in %.*s",
+			      "Cannot assign built-in procedure '%s' to %.*s%.*s",
 			      expr_str,
+			      LIT(article),
 			      LIT(context_name));
 			break;
 		case Addressing_Type:
 			if (is_type_polymorphic(operand->type)) {
 				error(operand->expr,
-				      "Cannot assign '%s' which is a polymorphic type in %.*s",
+				      "Cannot assign '%s' — a polymorphic type — to %.*s%.*s",
```

Comment:
I recommend using a hyphen `-`, even if an em dash is more correct. And if you were going to be very correct, you shouldn't have any space around the em dash either.

Reply:
> And if you were going to be very correct, you shouldn't have any space around the em dash either.

I'm aware and tried it. It looks worse.

Reply:
Just use commas instead? e.g.
```
Cannot assign '%s', a polymorphic type, to %.*s%.*s
```

Reply:
I just changed it to `-` at your recommendation?